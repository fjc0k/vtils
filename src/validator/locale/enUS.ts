/* istanbul ignore file */
import { yup } from '../yup.js'

// ref: https://github.com/jquense/yup/blob/master/src/locale.js

export const enUS: yup.Locale = {
  mixed: {
    default: '${path} is invalid',
    required: '${path} is a required field',
    oneOf: '${path} must be one of the following values: ${values}',
    notOneOf: '${path} must not be one of the following values: ${values}',
    defined: '${path} must be defined',
    notType: ({ path, type, value, originalValue }) => {
      const isCast = originalValue != null && originalValue !== value
      const msg = [
        `${path} must be a \`${type}\` type, `,
        `but the final value was: \`${yup.printValue(value, true)}\``,
        !isCast
          ? '.'
          : ` (cast from the value \`${yup.printValue(
              originalValue,
              true,
            )}\`).`,
        value !== null
          ? ''
          : '\n If "null" is intended as an empty value be sure to mark the schema as `.nullable()`',
      ]
        .filter(Boolean)
        .join('')
      return msg
    },
  },
  string: {
    length: '${path} must be exactly ${length} characters',
    min: '${path} must be at least ${min} characters',
    max: '${path} must be at most ${max} characters',
    matches: '${path} must match the following: "${regex}"',
    email: '${path} must be a valid email',
    url: '${path} must be a valid URL',
    trim: '${path} must be a trimmed string',
    lowercase: '${path} must be a lowercase string',
    uppercase: '${path} must be a upper case string',
    chineseIDCardNumber: '${path} must be a Chinese identity card number',
    chineseMobilePhoneNumber: '${path} must be a Chinese mobile phone number',
  },
  number: {
    min: '${path} must be greater than or equal to ${min}',
    max: '${path} must be less than or equal to ${max}',
    lessThan: '${path} must be less than ${less}',
    moreThan: '${path} must be greater than ${more}',
    positive: '${path} must be a positive number',
    negative: '${path} must be a negative number',
    integer: '${path} must be an integer',
    id: '${path} must be a positive integer',
    positiveInteger: '${path} must be a positive integer',
    negativeInteger: '${path} must be a negative integer',
    nonPositive: '${path} must be a non-positive number',
    nonNegative: '${path} must be a non-negative number',
    nonPositiveInteger: '${path} must be a non-positive integer',
    nonNegativeInteger: '${path} must be a non-negative integer',
  },
  date: {
    min: '${path} field must be later than ${min}',
    max: '${path} field must be at earlier than ${max}',
  },
  boolean: {},
  object: {
    noUnknown: '${path} field has unspecified keys: ${unknown}',
  },
  array: {
    min: '${path} field must have at least ${min} items',
    max: '${path} field must have less than or equal to ${max} items',
  },
}
