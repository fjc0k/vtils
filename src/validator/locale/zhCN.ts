/* istanbul ignore file */
import { yup } from '../yup'

export const zhCN: yup.Locale = {
  mixed: {
    default: ({ label }) => `${label || '此项'}错误`,
    required: ({ label }) => `${label || '此项'}必填`,
    oneOf: ({ label, values }) =>
      `${label || '此项'}必须是下列值之一: ${values}`,
    notOneOf: ({ label, values }) =>
      `${label || '此项'}必须不是下列值之一: ${values}`,
    defined: ({ label }) => `${label || '此项'}必须已定义`,
    notType: ({ label, type }) =>
      `${label || '此项'}必须是一个${
        type === 'number'
          ? '数字'
          : type === 'string'
          ? '字符串'
          : type === 'boolean'
          ? '布尔值'
          : type === 'object'
          ? '对象'
          : type === 'array'
          ? '数组'
          : type === 'date'
          ? '日期'
          : ` ${type} 类型的值`
      }`,
  },
  string: {
    length: ({ label, length }) =>
      `${label || '此项'}必须仅包含${length}位字符`,
    min: ({ label, min }) => `${label || '此项'}至少应包含${min}位字符`,
    max: ({ label, max }) => `${label || '此项'}最多可包含${max}位字符`,
    matches: ({ label, regex }) =>
      `${label || '此项'}必须匹配该正则表达式: ${regex}`,
    email: ({ label }) => `${label || '此项'}必须是一个合法的邮箱`,
    url: ({ label }) =>
      `${label || '此项'}必须是一个以 http:// 或 https:// 开头的网络链接`,
    trim: ({ label }) => `${label || '此项'}必须是一个没有旁白的字符串`,
    lowercase: ({ label }) => `${label || '此项'}必须是一个小写的字符串`,
    uppercase: ({ label }) => `${label || '此项'}必须是一个大写的字符串`,
    chineseMobilePhoneNumber: ({ label }) =>
      `${label || '此项'}必须是一个合法的手机号码`,
    chineseIDCardNumber: ({ label }) =>
      `${label || '此项'}必须是一个合法的身份证号码`,
  },
  number: {
    min: ({ label, min }) => `${label || '此项'}必须大于或等于${min}`,
    max: ({ label, max }) => `${label || '此项'}必须小于或等于${max}`,
    lessThan: ({ label, less }) => `${label || '此项'}必须小于${less}`,
    moreThan: ({ label, more }) => `${label || '此项'}必须大于${more}`,
    positive: ({ label }) => `${label || '此项'}必须是一个正数`,
    negative: ({ label }) => `${label || '此项'}必须是一个负数`,
    integer: ({ label }) => `${label || '此项'}必须是一个整数`,
    id: ({ label }) => `${label || '此项'}必须是一个正整数`,
  },
  date: {
    min: ({ label, min }) => `${label || '此项'}必须晚于${min}`,
    max: ({ label, max }) => `${label || '此项'}必须早于${max}`,
  },
  boolean: {},
  object: {
    noUnknown: ({ label, unknown }) =>
      `${label || '此项'}上有未定义的键: ${unknown}`,
  },
  array: {
    min: ({ label, min }) => `${label || '此项'}至少应包含${min}个元素`,
    max: ({ label, max }) => `${label || '此项'}最多可包含${max}个元素`,
  },
}
