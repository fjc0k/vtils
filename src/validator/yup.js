import { zhCN } from './locale/zhCN'
import * as yup from './yupSource'

// 设置中文为默认语言
yup.setLocale(zhCN)

export { yup }
