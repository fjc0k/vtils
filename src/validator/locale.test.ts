import { enUS } from './locale/enUS.ts'
import { getZhCN, zhCN } from './locale/zhCN.ts'
import { yup } from './yup.js'

describe('locale', () => {
  for (const [name, locale] of Object.entries({ enUS, zhCN })) {
    test(name, () => {
      expect(Object.keys(locale)).toEqual([
        'mixed',
        'string',
        'number',
        'date',
        'boolean',
        'object',
        'array',
      ])
    })
  }

  test('中文', async () => {
    await expect(yup.number().integer().validate(9.2)).rejects.toThrowError(
      /此项必须是一个整数/,
    )
    await expect(
      yup.number().label('年龄').integer().validate(9.2),
    ).rejects.toThrowError(/年龄必须是一个整数/)
  })

  test('中文-自定义label', async () => {
    yup.setLocale(
      getZhCN({
        getLabel: params => params.label || params.path,
      }),
    )
    await expect(
      yup
        .object({
          id: yup.number(),
        })
        .validate({
          // @ts-expect-error
          id: 't',
        }),
    ).rejects.toThrowErrorMatchingSnapshot()
  })
})
