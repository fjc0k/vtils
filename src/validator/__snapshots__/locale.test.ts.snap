// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locale 中文-自定义label 1`] = `"id必须是一个数字"`;
